### PR TITLE
use informerfactory start all informers

### PR DESCRIPTION
**Pull request by lucming**
if there are many informer,we'd better use informerfactory to operate all informers,can make code more clearly.

### CODE REVIEW

File: pkg/controllers/job/job_controller.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"fmt"
 	"hash"
 	"hash/fnv"
+	"k8s.io/client-go/informers"
```
Comment (jasonliu747) [inline pkg/controllers/job/job_controller.go:23]: please format imports

Comment (lucming) [reply pkg/controllers/job/job_controller.go]: > please format imports

done :)

File: pkg/controllers/queue/queue_controller.go [hunk 2]
```diff
@@ -171,14 +175,9 @@ func (c *queuecontroller) Run(stopCh <-chan struct{}) {
...
-	go c.queueInformer.Informer().Run(stopCh)
-	go c.pgInformer.Informer().Run(stopCh)
-	go c.cmdInformer.Informer().Run(stopCh)
+	c.vcInformerFactory.Start(stopCh)
 
-	if !cache.WaitForCacheSync(stopCh, c.queueSynced, c.pgSynced, c.cmdSynced) {
-		klog.Errorf("unable to sync caches for queue controller.")
-		return
-	}
+	c.vcInformerFactory.WaitForCacheSync(stopCh)
```
[the remark this answers is not in the document]
Comment (lucming) [reply pkg/controllers/queue/queue_controller.go]: > Please take care of return value of 'WaitForCacheSync'.

okay, let me deal with the return value,thanks

File: pkg/controllers/job/job_controller.go [hunk 3]
```diff
@@ -219,17 +227,11 @@ func (cc *jobcontroller) Initialize(opt *framework.ControllerOption) error {
...
-	go cc.pvcInformer.Informer().Run(stopCh)
-	go cc.pgInformer.Informer().Run(stopCh)
-	go cc.svcInformer.Informer().Run(stopCh)
-	go cc.cmdInformer.Informer().Run(stopCh)
-	go cc.pcInformer.Informer().Run(stopCh)
-	go cc.queueInformer.Informer().Run(stopCh)
-
-	cache.WaitForCacheSync(stopCh, cc.jobSynced, cc.podSynced, cc.pgSynced,
-		cc.svcSynced, cc.cmdSynced, cc.pvcSynced, cc.pcSynced, cc.queueSynced)
+	cc.informerFactory.Start(stopCh)
```
[the remark this answers is not in the document]
Comment (lucming) [reply pkg/controllers/job/job_controller.go:230]: `// Start initializes all requested informers.
func (f *sharedInformerFactory) Start(stopCh <-chan struct{}) {
	f.lock.Lock()
	defer f.lock.Unlock()

	for informerType, informer := range f.informers {
		if !f.startedInformers[informerType] {
			go informer.Run(stopCh)
			f.startedInformers[informerType] = true
		}
	}
}`

[the remark this answers is not in the document]
Comment (lucming) [reply pkg/controllers/job/job_controller.go:230]: > How about the performance? Previously there are multiple go routines for informers like jobInformer, podInformer, pvcInformer etc.

`
// Start initializes all requested informers.
func (f *sharedInformerFactory) Start(stopCh <-chan struct{}) {
	f.lock.Lock()
	defer f.lock.Unlock()

	for informerType, informer := range f.informers {
		if !f.startedInformers[informerType] {
			go informer.Run(stopCh)
			f.startedInformers[informerType] = true
		}
	}
}
`
the same logic with the old code

**Comment by lucming** [comment]
Is there anything else that needs to be changed?

File: pkg/controllers/queue/queue_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (william-wang) [reply pkg/controllers/queue/queue_controller.go]: Sounds good now.

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**